src/actions/DataPanelAction.java~

```diff
@@ -0,0 +1,45 @@
+package actions;
+
+import javax.swing.*;
+
+import static controller.Controller.updateDataTable;
+import static controller.Main.*;
+import static model.Operations.*;
+import static view.Listener.saveFlag;
+import static view.MyFrame.stpFrame;
+
+/*
+ * Created by Ahmed on 14-Aug-16.
+ */
+public class DataPanelAction {
+
+    public void updateTableButtonAction() {
+        updateDataTable();
+    }
+
+    public void exitButtonAction() {
+        int toSave;
+        if (saveFlag) {
+            toSave = JOptionPane.showConfirmDialog(stpFrame, "Unsaved work !\n" +
+                    "Do you want to save it ?", "Warning", JOptionPane.OK_CANCEL_OPTION);
+            if (toSave == 0) saveButtonAction();
+        }
+        System.exit(0);
+    }
+
+    public void saveButtonAction() {
+        setFileName(CREATED_CSV);
```

Comment:
Why and how u used CREATED_CSV ? 
it's supposed u fixed it and now we take the file from a JFileChooser